File: 프로그래머스/2/42578. 의상/의상.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(string[,] clothes) {
        var countType = new Dictionary<string, int>();
        int row = clothes.GetLength(0);
        
        for(int i = 0; i < row; i++)
        {
            string type = clothes[i, 1];
            if(countType.ContainsKey(type)) countType[type]++;
            else countType[type] = 1;
        }
        
        int answer = 1;
        foreach(var kv in countType)
        {
            answer *= (kv.Value + 1);
        }
        answer = answer - 1;
        return answer;
    }
}

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map<string, int> mp;
    
    for(int i = 0; i < clothes.size(); i++)
    {
        mp[clothes[i][1]]++;
    }
    
    for(auto it : mp)
    {
        answer *= (it.second + 1);
    }
    
    return answer - 1;
}